fetch.js

```diff
@@ -105,9 +105,34 @@
     this.bodyUsed = false
 
     if (blobSupport) {
+      this._initBody = function(body) {
+        this._bodyInit = body
+        if (typeof body === 'string') {
+          this._bodyText = body
+        } else if (Blob.prototype.isPrototypeOf(body)) {
+          this._bodyBlob = body
+        } else if (FormData.prototype.isPrototypeOf(body)) {
+          this._bodyFormData = body
+        } else if (!body) {
+          this._bodyText = ''
+        } else {
+          throw new Error('unsupported BodyInit type')
+        }
+      }
+
       this.blob = function() {
         var rejected = consumed(this)
-        return rejected ? rejected : Promise.resolve(this._bodyBlob)
+        if (rejected) {
+          return rejected
+        }
+
+        if (this._bodyBlob) {
+          return Promise.resolve(this._bodyBlob)
+        } else if (this._bodyFormData) {
+          throw new Error('could not read FormData body as blob')
```

Comment:
In the polyfill case, if you construct a Request/Reponse body with FormData then ask for the blob, we just crash :/

```
new Response(new FormData()).blob()
// throws an error
```